x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/authc/ApiKeyIntegTests.java

```diff
@@ -1497,71 +1495,73 @@ public void testUpdateApiKey() throws ExecutionException, InterruptedException,
         }
     }
 
-    private List<RoleDescriptor> randomRoleDescriptors() {
-        int caseNo = randomIntBetween(0, 2);
-        return switch (caseNo) {
-            case 0 -> List.of(new RoleDescriptor(randomAlphaOfLength(10), new String[] { "all" }, null, null));
-            case 1 -> List.of(
-                new RoleDescriptor(randomAlphaOfLength(10), new String[] { "all" }, null, null),
-                RoleDescriptorTests.randomRoleDescriptor()
-            );
-            case 2 -> null;
-            default -> throw new IllegalStateException("unexpected case no");
-        };
-    }
-
     public void testUpdateApiKeyNotFoundScenarios() throws ExecutionException, InterruptedException {
-        final Tuple<CreateApiKeyResponse, Map<String, Object>> createdApiKey = createApiKey(ES_TEST_ROOT_USER, null);
+        final Tuple<CreateApiKeyResponse, Map<String, Object>> createdApiKey = createApiKey(TEST_USER_NAME, null);
```

Comment:
Switching users here because the role of `ES_TEST_ROOT_USER` does not pass validation that happens during role parsing under the hood of `expectRoleDescriptorsForApiKey`

Reply:
This is odd: I thought the leading `_` in the role name (`_es_test_root`) was failing the validation but that's not the case apparently. I think the original failure was due to the random role descriptors being generated failing for a different reason which I have since fixed. 

**tl;dr** it was actually fine to use the root user here. I'll keep `TEST_USER_NAME` here since the code is already rewritten to use that name. Sorry for the confusion!

Reply:
Thanks for the explanation. All good.